libpod/container_internal_linux.go

```diff
@@ -48,23 +50,55 @@ func (c *Container) unmountSHM(mount string) error {
 // prepare mounts the container and sets up other required resources like net
 // namespaces
 func (c *Container) prepare() (err error) {
+	var (
+		wg                              sync.WaitGroup
+		netNS                           ns.NetNS
+		networkStatus                   []*cnitypes.Result
+		createNetNSErr, mountStorageErr error
+		mountPoint                      string
+	)
+
+	wg.Add(2)
+
+	go func() {
+		defer wg.Done()
+		// Set up network namespace if not already set up
+		if c.config.CreateNetNS && c.state.NetNS == nil && !c.config.PostConfigureNetNS {
+			netNS, networkStatus, createNetNSErr = c.runtime.createNetNS(c)
+		}
+	}()
 	// Mount storage if not mounted
-	if err := c.mountStorage(); err != nil {
-		return err
-	}
+	go func() {
+		defer wg.Done()
+		mountPoint, mountStorageErr = c.mountStorage()
+	}()
 
-	// Set up network namespace if not already set up
-	if c.config.CreateNetNS && c.state.NetNS == nil && !c.config.PostConfigureNetNS {
-		if err := c.runtime.createNetNS(c); err != nil {
-			// Tear down storage before exiting to make sure we
-			// don't leak mounts
-			if err2 := c.cleanupStorage(); err2 != nil {
-				logrus.Errorf("Error cleaning up storage for container %s: %v", c.ID(), err2)
-			}
-			return err
+	wg.Wait()
+	if createNetNSErr != nil {
+		if mountStorageErr != nil {
```

Comment:
Why is this error handling condition nested?

Reply:
bc @mheon wanted to deal with the error messages in conjunction with each other.